test/gtest/common/test.cc

```diff
@@ -342,7 +342,7 @@ void test_base::TearDownProxy() {
 
     watchdog_signal();
 
-    if (m_initialized) {
+    if (m_initialized || (m_state == SKIPPED)) {
```

Comment:
Now I see what you are talking about, thanks for your comment.

This changes causes errors in many tests since `cleanup()` started to be called even when initialization not even started, because test was skipped during `check_skip_test()` execution. This happens for tests that use `UCS_TEST_SKIP_COND_P` macro.

So it seems like this change is not logically correct. @yosefe should we try to introduce some new states in addition to existing one to be able to distinguish skip happened during `init()` and skip happened during `check_skip_test()` or we can leave thing as they are now?